hstream/src/HStream/Server/Config.hs

```diff
@@ -156,11 +179,12 @@ parseJSONToOptions CliOptions {..} obj = do
   nodeId      <- nodeCfgObj .:  "id"
   nodeAddress <- nodeCfgObj .:  "address"
   nodePort    <- nodeCfgObj .:? "port" .!= 6570
-  nodeInternalPort <- nodeCfgObj .:? "internal-port" .!= 6571
-  zkuri            <- nodeCfgObj .:  "zkuri"
-  recordCompression   <- read <$> nodeCfgObj .:? "compression" .!= "lz4"
-  nodeLogLevel     <- nodeCfgObj .:? "log-level" .!= "info"
-  nodeLogWithColor <- nodeCfgObj .:? "log-with-color" .!= True
+  nodeInternalPort  <- nodeCfgObj .:? "internal-port" .!= 6571
+  zkuri             <- nodeCfgObj .:  "zkuri"
+  recordCompression <- read <$> nodeCfgObj .:? "compression" .!= "lz4"
+  nodeLogLevel      <- nodeCfgObj .:? "log-level" .!= "info"
+  nodeLogWithColor  <- nodeCfgObj .:? "log-with-color" .!= True
+  _maxRecordSize    <- nodeCfgObj .:? "max-record-size" .!= (1048576 - 1024)
```

Comment:
1. Use a more readable format (e.g. `1024*1024 - 1024`, also comment why minus 1024)
2. Comment the default value details in the hstream config file
3. Make sure the logdevice can accept this record